SpatialGDK/Source/SpatialGDKTests/SpatialGDK/RPCService/RPCServiceTest.cpp

```diff
@@ -0,0 +1,568 @@
+
+	// Send RPCs to the point where we will overflow
+	int RPCsToSend = static_cast<int>(GetDefault<USpatialGDKSettings>()->GetRPCRingBufferSize(ERPCType::ClientReliable));
+	for (int i = 0; i < RPCsToSend; ++i)
+	{
+		RPCService.PushRPC(RPCTestEntityId_1, ERPCType::ClientUnreliable, SimplePayload);
+	}
+
+	SpatialGDK::EPushRPCResult Result = RPCService.PushRPC(RPCTestEntityId_1, ERPCType::ClientUnreliable, SimplePayload);
+	bool bTestPassed = Result == SpatialGDK::EPushRPCResult::DropOverflowed;
+	TestTrue("Push RPC returned expected results", bTestPassed);
+	return bTestPassed;
+}
+
+RPC_SERVICE_TEST(GIVEN_authority_over_client_endpoint_WHEN_push_overflow_client_reliable_rpcs_to_the_service_THEN_check_push_results)
+{
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService({ RPCTestEntityId_1 }, CLIENT_AUTH);
+
+	// Send RPCs to the point where we will overflow
+	int RPCsToSend = static_cast<int>(GetDefault<USpatialGDKSettings>()->GetRPCRingBufferSize(ERPCType::ServerReliable));
+	for (int i = 0; i < RPCsToSend; ++i)
+	{
+		RPCService.PushRPC(RPCTestEntityId_1, ERPCType::ServerReliable, SimplePayload);
+	}
+
+	SpatialGDK::EPushRPCResult Result = RPCService.PushRPC(RPCTestEntityId_1, ERPCType::ServerReliable, SimplePayload);
+	bool bTestPassed = Result == SpatialGDK::EPushRPCResult::QueueOverflowed;
+	TestTrue("Push RPC returned expected results", bTestPassed);
+	return bTestPassed;
+}
+
+RPC_SERVICE_TEST(GIVEN_authority_over_client_endpoint_WHEN_push_overflow_client_unreliable_rpcs_to_the_service_THEN_check_push_results)
+{
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService({ RPCTestEntityId_1 }, CLIENT_AUTH);
+
+	// Send RPCs to the point where we will overflow
+	int RPCsToSend = static_cast<int>(GetDefault<USpatialGDKSettings>()->GetRPCRingBufferSize(ERPCType::ServerReliable));
+	for (int i = 0; i < RPCsToSend; ++i)
+	{
+		RPCService.PushRPC(RPCTestEntityId_1, ERPCType::ServerUnreliable, SimplePayload);
+	}
+
+	SpatialGDK::EPushRPCResult Result = RPCService.PushRPC(RPCTestEntityId_1, ERPCType::ServerUnreliable, SimplePayload);
+	bool bTestPassed = Result == SpatialGDK::EPushRPCResult::DropOverflowed;
+	TestTrue("Push RPC returned expected results", bTestPassed);
+	return bTestPassed;
+}
+
+RPC_SERVICE_TEST(GIVEN_authority_over_server_endpoint_WHEN_push_overflow_multicast_rpcs_to_the_service_THEN_check_push_results)
+{
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService({ RPCTestEntityId_1 }, SERVER_AUTH);
+
+	// Send RPCs to the point where we will overflow
+	int RPCsToSend = static_cast<int>(GetDefault<USpatialGDKSettings>()->GetRPCRingBufferSize(ERPCType::NetMulticast));
+	for (int i = 0; i < RPCsToSend; ++i)
+	{
+		RPCService.PushRPC(RPCTestEntityId_1, ERPCType::NetMulticast, SimplePayload);
+	}
+
+	SpatialGDK::EPushRPCResult Result = RPCService.PushRPC(RPCTestEntityId_1, ERPCType::NetMulticast, SimplePayload);
+	bool bTestPassed = Result == SpatialGDK::EPushRPCResult::Success;
```

Comment:
Overflowing unreliable rpcs results in DropOverflowed, but overflowing multicast results in Success? Probably not a question for this PR, just curious.